generate_instructions.py

```diff
@@ -32,10 +32,13 @@ def __init__(self, side1_letter,
         self.block_id = block_id if block_id > 0 else Block.get_next_block_id()
 
     def __eq__(self, other):
-        return self.block_id == other.block_id
+        return type(other) == type(self) and self.block_id == other.block_id
 
-    def __ne__(self, other):
-        return not self == other
```

Comment:
Why delete `__ne__`? I can only see that becoming a subtle problem in the future.

Reply:
I believe in the absence of `__ne__`, inequality checks become `not __eq__` - why risk `__eq__` and `__ne__` getting out of sync?

Reply:
https://stackoverflow.com/questions/4352244/python-should-i-implement-ne-operator-based-on-eq

Reply:
Yeah he pointed that out to me last week or something.